widget/tabcontainer.go

```diff
@@ -122,12 +122,21 @@ func (c *TabContainer) Remove(item *TabItem) {
 
 // RemoveIndex removes tab by index
 func (c *TabContainer) RemoveIndex(index int) {
+	if c.current == index {
+		c.setTabIndex(-1)
+	}
+	if c.current > index {
+		c.setTabIndex(c.current - 1)
+	}
```

Comment:
If either of these branches are taken, the TabContainer will be refreshed, and then when the items are set below the TabContainer will be refreshed again. It might be better to assign the field directly;
```suggestion
	if c.current == index {
		c.current = -1
	}
	if c.current > index {
		c.current = c.current - 1
	}
```

Reply:
I was considering doing that but in the original SelectTabIndex after the Refresh it does
`	if c.OnChanged != nil {
		c.OnChanged(c.Items[c.current])
	}`
I am unsure what that does, so thought it safer to include it at the cost of an extra refresh.  I could have just cut/pasted it everywhere but did not think that was the correct way. So I split SelectTabIndex to create setTabIndex and called that to only have the functionality in one place. 
I am more than happy to make your suggested change if you think the lines above are not needed when current=-1

Reply:
OnChanged is a field to which the developer can assign a function if they want to respond when the current tab changes.

You bring up a good point. @andydotxyz should devs expect a change notification after calling RemoveIndex, or SetItems? Would they expect `OnChange(nil)` to be called when there is no longer a selection ie. `current == -1`? Perhaps it would be better if TabContainer used the same OnSelected, OnUnselected pattern as the collections.

One downside I can see in RemoveIndex is that the change notification will be triggered _before_ the index is removed, so OnChange will be called with the wrong TabItem.

Another issues is that setTabIndex doesn't have the bounds checking so it could crash when indexing the Items slice with -1

Reply:
I would expect the OnChange to fire when the selected item changes. If the item removed after current tab then not firing, if the removal was before then it's tough to say, perhaps not, as semantically it's still the same content?

If we set the index to -1 (and it wasn't before) then we do need to fire yes.
Moving to OnSelected could make sense, I think it's on (or should be on) the consistent-selection-api ticket #1395.